upcloud/service/storage_test.go

```diff
@@ -376,3 +376,63 @@ func TestDirectUploadStorageImport(t *testing.T) {
 		require.Equal(t, sha256sum, afterStorageImportDetails.SHA256Sum)
 	})
 }
+
+// TestResizeStorage performs the following actions:
+// - creates a server
+// - stops the server
+// - detaches the storage
+// - resizes the storage disk
+// - attaches the storage
+// - resizes the storage
+// - cleanup
+func TestResizeStorage(t *testing.T) {
+	t.Parallel()
+
+	record(t, "resizestorage", func(t *testing.T, rec *recorder.Recorder, svc *Service) {
+		// start server
+		serverDetails, err := createMinimalServer(svc, "TestResizeStorage")
+		require.NoError(t, err)
+
+		// stop server
+		require.NoError(t, stopServer(svc, serverDetails.UUID))
+
+		// detach storage
```

Comment:
Resize should be possible without detaching, too

Reply:
I think that storage needs to be detached before modification (increase disk size). Resizing partition/fs can be done to attached storage (line 425).